### PR TITLE
add: CreateALabelOnBoardTest

### PR DESCRIPTION
**Pull request by KhraschukZahar**

### CODE REVIEW

**Review by ArtemKostiuktest — APPROVED** [review]

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
 import io.restassured.common.mapper.TypeRef;
+import io.restassured.response.Response;
 
 import java.util.List;
+import java.util.Map;
 
 import static io.restassured.RestAssured.given;
 import static java.net.HttpURLConnection.HTTP_OK;
 
+
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java]: delete empty space

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 2]
```diff
@@ -1,29 +1,51 @@
...
+                    .spec(requestSpec)
+                    .when()
+                    .post("1/labels")
+                    .then()
+                    .statusCode(statusCode)
+                    .extract()
+                    .response();
+        }
+
+        public List<CreateLabelResponse> getLabelOnABoard (String boardId){
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java]: rename

File: src/test/java/api/post/CreateALabelOnBoardTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @BeforeMethod
+    public void createBoard() {
+        boardId = boardRestTestClient.createNewBoard(constructDefaultBoardKeyValue()).getId();
+    }
+
+    @Test
+    public void createLabelOnBoard() {
+         labelId = labelRestTestClient.createLabel(constructDefaultListKeyValue("label1","blue",boardId)).getId();
+         List<CreateLabelResponse> LabelDataResponse = labelRestTestClient.getLabelOnABoard(boardId);
```
Comment (SoniaMiach) [inline src/test/java/api/post/CreateALabelOnBoardTest.java]: add empty space

File: src/test/java/api/post/CreateALabelOnBoardTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+         Assert.assertNotNull(labelId, "Label has not been created.");
+         Assert.assertFalse(LabelDataResponse.isEmpty(), "The list of labels on the board is empty.");
```
Comment (Goldbe54) [inline src/test/java/api/post/CreateALabelOnBoardTest.java]: Wrong assert. Try to check the label from the list of responses. Does it contain the correct name and color

File: src/test/java/api/get/GetALabelsOnBoardTest.java [hunk 5]
```diff
@@ -22,7 +22,7 @@ public void createBoard() {
 
     @Test
     public void getLabelsOnBoard() {
-        List<String> labelNames = labelRestTestClient.getLabelOnABoard(boardId).stream().map(CreateLabelResponse::getColor).collect(Collectors.toList());
+        List<String> labelNames = labelRestTestClient.getLabelsOnABoard(boardId).stream().map(CreateLabelResponse::getColor).collect(Collectors.toList());
         List<String> expectedLabelColors = List.of("blue", "green", "orange", "purple", "red", "yellow");
         Assert.assertFalse(labelNames.isEmpty(), "Label list is empty.");
```
Comment (SoniaMiach) [inline src/test/java/api/get/GetALabelsOnBoardTest.java]: add empty space

**Review by SoniaMiach — APPROVED** [review]

**Review by SoniaMiach — APPROVED** [review]

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 6]
```diff
@@ -2,28 +2,62 @@
...
+                    .when()
+                    .log().all()
+                    .get("/1/boards/{id}/labels", boardId)
+                    .then()
+                    .statusCode(HTTP_OK)
+                    .extract()
+                    .as(new TypeRef<List<CreateLabelResponse>>() {
+                    });
+        }
+    public CreateLabelResponse getLabel (String labelId){
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java]: Separate sections with a space

**Review by Goldbe54 — COMMENTED** [review]
Merge curent master branch

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 7]
```diff
@@ -1,20 +1,44 @@
...
+import io.restassured.response.Response;
+import org.apache.commons.lang3.RandomStringUtils;
 
 import java.util.List;
+import java.util.Map;
 
 import static io.restassured.RestAssured.given;
 import static java.net.HttpURLConnection.HTTP_OK;
 
+
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java]: delete empty space. Again

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 8]
```diff
@@ -1,20 +1,44 @@
...
+                .spec(requestSpec)
+                .when()
+                .post("1/labels")
+                .then()
+                .statusCode(statusCode)
+                .extract()
+                .response();
+    }
+
+    @Step("Extracting information about all label by id: {boardId}")
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java]: "..all labels by board id"

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 9]
```diff
@@ -26,4 +50,23 @@ public List<CreateLabelResponse> getLabelOnABoard(String boardId) {
                 .as(new TypeRef<List<CreateLabelResponse>>() {
                 });
     }
+
+    @Step("Extracting information about label by id: {boardId}")
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java]: "...by board id"

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 10]
```diff
@@ -26,4 +50,23 @@ public List<CreateLabelResponse> getLabelOnABoard(String boardId) {
...
+                .extract().as(CreateLabelResponse.class);
+    }
+
+    public static Map<String, String> constructDefaultListKeyValue(String labelName, String labelColor, String idBoard) {
+        return Map.of("name", labelName, "color", labelColor, "idBoard", idBoard);
+    }
+
+    public static Map<String, String> constructDefaultBoardKeyValue() {
+        return Map.of("name", "Test board " + RandomStringUtils.randomAlphanumeric(3));
+    }
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java]: replace to BoardRestTestClient

File: src/test/java/api/get/GetALabelsOnBoardTest.java [hunk 11]
```diff
@@ -22,19 +23,18 @@ public void createBoard() {
...
-        List<String> labelNames = labelRestTestClient.getLabelOnABoard(boardId).stream().map(CreateLabelResponse::getColor).collect(Collectors.toList());
+        List<String> labelNames = labelRestTestClient.getLabelsOnABoard(boardId).stream().map(CreateLabelResponse::getColor).collect(Collectors.toList());
         List<String> expectedLabelColors = List.of("blue", "green", "orange", "purple", "red", "yellow");
+
         Assert.assertFalse(labelNames.isEmpty(), "Label list is empty.");
         Assert.assertTrue(labelNames.containsAll(expectedLabelColors), "Not all expected label colors are found.");
     }
 
     @AfterMethod
+    @Step("Delete the test board")
```
Comment (Goldbe54) [inline src/test/java/api/get/GetALabelsOnBoardTest.java]: Try use "@Step" in BoardRestTestClient.java, upster deleteBoardIfExist method

File: src/test/java/api/get/GetALabelsOnBoardTest.java [hunk 12]
```diff
@@ -22,19 +23,18 @@ public void createBoard() {
...
 
     @Test(description = "AS2-15")
-    @Description("Get Labels on a Board")
+    @Description("Get labels on a board")
     public void getLabelsOnBoard() {
-        List<String> labelNames = labelRestTestClient.getLabelOnABoard(boardId).stream().map(CreateLabelResponse::getColor).collect(Collectors.toList());
+        List<String> labelNames = labelRestTestClient.getLabelsOnABoard(boardId).stream().map(CreateLabelResponse::getColor).collect(Collectors.toList());
         List<String> expectedLabelColors = List.of("blue", "green", "orange", "purple", "red", "yellow");
+
         Assert.assertFalse(labelNames.isEmpty(), "Label list is empty.");
```
Comment (Goldbe54) [inline src/test/java/api/get/GetALabelsOnBoardTest.java]: Delete this Assert. There is no point in checking if the label is empty. Write an assertion that checks whether the expected number of labels corresponds to the actual number

File: src/test/java/api/clients/LabelRestTestClient.java [hunk 13]
```diff
@@ -1,20 +1,44 @@
...
         super(url);
     }
 
-    public List<CreateLabelResponse> getLabelOnABoard(String boardId) {
+    public CreateLabelResponse createLabel(Map<String, String> labelKeyValue) {
+        return createLabelResponse(labelKeyValue, HTTP_OK)
+                .as(CreateLabelResponse.class);
+    }
+
+    @Step("Creating a new label")
```
Comment (Goldbe54) [inline src/test/java/api/clients/LabelRestTestClient.java:26]: Use "@Step" in "createLabel" method

File: src/test/java/api/post/CreateALabelOnBoardTest.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        labelId = labelRestTestClient.createLabel(constructDefaultListKeyValue(name, color, boardId)).getId();
+
+        CreateLabelResponse response = labelRestTestClient.getLabel(labelId);
+
+        Assert.assertEquals(name, response.getName());
+        Assert.assertEquals(color, response.getColor());
+    }
+
+    @AfterMethod
+    @Step("Delete the test board")
```
Comment (Goldbe54) [inline src/test/java/api/post/CreateALabelOnBoardTest.java]: Try use "Step" in BoardRestTestClient.java, upster deleteBoardIfExist method

File: src/test/resources/suites/khraschukSuite.xml [hunk 15]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="AllTestSuite">
+    <test name="getTest">
+        <classes>
+            <class name="api.get.GetALabelsOnBoardTest" />
+        </classes>
+    </test>
+    <test name="postTests">
+        <classes>
+            <class name="api.post.CreateALabelOnBoardTest" />
```
Comment (Goldbe54) [inline src/test/resources/suites/khraschukSuite.xml]: Make it like in master

**Review by Goldbe54 — APPROVED** [review]

**Pull request merged**